.travis.yml

```diff
@@ -13,20 +13,18 @@ branches:
 env:
   # add target platforms to build matrix
   # lint is a target so we don't need to lint multiple times
-  - TARGET='build_all PLATFORM=x86'
+  - TARGET='build_all PLATFORM=x86 CC=gcc-6 LD=gcc-6'
   - TARGET='build_all PLATFORM=stm32f0xx'
-  - TARGET='test_all PLATFORM=x86'
+  - TARGET='test_all PLATFORM=x86 CC=gcc-6 LD=gcc-6'
   - TARGET=lint
 
-before_install:
-  # update the libraries (ms-common, etc.)
-  - git submodule update --init --recursive
-
 before_script:
-  # install GCC ARM from GNU ARM Embedded Toolchain PPA
+  # install GCC ARM from GNU ARM Embedded Toolchain PPA and GCC 6
   - sudo add-apt-repository ppa:team-gcc-arm-embedded/ppa -y
+  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
```

Comment:
I think this is actually one of Travis' [whitelisted apt-sources](https://github.com/travis-ci/apt-source-whitelist/blob/2b5b082b09878010bd06e5293750fa79002d2b5c/ubuntu.json#L378), so we should probably be using those.

It should be a matter of just adding

```yaml
addons:
  apt:
    sources:
    # add PPAs with more up-to-date toolchains
    - ubuntu-toolchain-r-test
  packages:
    # install toolchains
    - gcc-6
    - g++-6
```